diff --git a/cmake/system.cmake b/cmake/system.cmake
index 68911de..881a1b0 100644
--- a/cmake/system.cmake
+++ b/cmake/system.cmake
@@ -9,7 +9,18 @@ include("${PROJECT_SOURCE_DIR}/cmake/system_check.cmake")
 
 if(CMAKE_CROSSCOMPILING AND NOT DEFINED TARGET)
   # Detect target without running getarch
-  if (ARM64)
+  if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Emscripten")
+    # https://github.com/xianyi/OpenBLAS/issues/3640#issuecomment-1144029630
+    set(NOFORTRAN 1)
+    set(USE_THREAD 0)
+    set(ARCH "riscv64")
+    # set(CC "emcc")
+    # set(HOSTCC "gcc")
+    # "The choice of RISCV64 for the GENERIC build has no consequence for the
+    # resulting code, except that it avoids complications from inline assembly
+    # code that an x86_64 build would bring in, both use the same plain C sources
+    set(TARGET "RISCV64_GENERIC")
+  elseif (ARM64)
     set(TARGET "ARMV8")
   elseif(ARM)
     set(TARGET "ARMV7") # TODO: Ask compiler which arch this is
